.github/workflows/build-windows.yml

```diff
@@ -3,13 +3,10 @@ name: build-windows
 on:
   pull_request:
     types: [opened, synchronize, reopened, ready_for_review]
+    paths-ignore: ['**.md', doc/**, AUTHORS, .mailmap, .github/**, '!.github/workflows/build-windows.yml']
   push:
-    branches:
-      - master
-      - main
-      - canary
-      - v[0-9]+.x-staging
-      - v[0-9]+.x
+    branches: [master, main, canary, 'v[0-9]+.x-staging', 'v[0-9]+.x']
+    paths-ignore: ['**.md', doc/**, AUTHORS, .mailmap, .github/**, '!.github/workflows/build-windows.yml']
```

Comment:
it's annoying that we have to repeat so much :(

Reply:
I completely agree but to my knowledge there is no way to avoid these repetitions

Reply:
Does the `!` pattern work for `paths-ignore`? Docs suggest it only for `paths`: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths

Reply:
Yes it work, i tested it in my fork.